app/app_test.go

```diff
@@ -47,6 +53,45 @@ func TestNew(t *testing.T) {
 	})
 }
 
+func TestOfferSnapshot(t *testing.T) {
```

Comment:
Note: this unit test doesn't actually hit the `error` observed before the fix. To do that we need to start a channel and stream chunks of the snapshot so I may opt to do that in an integration test instead.